### ISSUE TITLE
iOS App Built with react-native Failing Unexpectedly with Xcode Task

### ISSUE THREAD

**Issue by woodrufs** [actions/virtual-environments#2332]
**Description**  
We are working with a react-native (v0.59.10) app that is build using the macOS 10.14 VM using Azure Pipelines. About a week ago, our pipeline broke unexpectedly. We had a successful build December 15 at 10:30 AM EST, but our next build December 16 at 4:56 PM failed. This was unexpected since our only change was a patch fix in a library we use. Upon inspecting the log, the error happened in our Xcode task. This was the error:
```
/usr/bin/xcodebuild -workspace /Users/runner/work/1/s/ios/PlantFloor.xcworkspace -scheme PlantFloor-Development archive -configuration Development -archivePath /Users/runner/work/1/s/ios/archive/dev/PlantFloor-Development CODE_SIGN_STYLE=Manual CODE_SIGN_IDENTITY=iPhone Distribution PROVISIONING_PROFILE= PROVISIONING_PROFILE_SPECIFIER=MUX Plant Floor (Temporary) | /usr/local/lib/ruby/gems/2.7.0/bin/xcpretty --no-color
▸ Building React/double-conversion [(Release)]
▸ Check Dependencies
▸ Running script 'Install Third Party'
** ARCHIVE FAILED **

The following build commands failed:
	PhaseScriptExecution Install\ Third\ Party /Users/runner/Library/Developer/Xcode/DerivedData/PlantFloor-eekvtuatgnpjwsebsuzvotnbadhy/Build/Intermediates.noindex/ArchiveIntermediates/PlantFloor-Development/IntermediateBuildFilesPath/React.build/Release-iphoneos/double-conversion.build/Script-190EE32F1E6A43DE00A8543A.sh
(1 failure)
```
I searched the repo for the "Install Third Party" script that is failing, and the only result is inside the node_modules folder: `/repo_root/node_modules/react-native/React/React.xcodeproj/project.pbxproj`

Our Xcode task (which hasn't changed in quite some time) looks like:
```
- task: Xcode@5
  displayName: 'Create iOS $(configuration) build'
  inputs:
    actions: clean
    configuration: $(configuration)
    sdk: ''
    xcWorkspacePath: ios/PlantFloor.xcworkspace
    scheme: $(scheme)
    packageApp: true
    archivePath: ios/archive/$(workPath)/
    exportPath: ios/output/$(workPath)/
    exportOptions: plist
    exportOptionsPlist: ${{ parameters.plistName }}
    signingOption: manual
    signingIdentity: 'iPhone Distribution'
    provisioningProfileUuid: '$(provisionUUID)'
    provisioningProfileName: ${{ parameters.profileName }}
    args: ''
    workingDirectory: ios
```

In between the time that our build succeeded and then failed, we have found that these tasks were upgraded on the build agent (macOS-10.14):
Task|Original Version|Updated Version
---|-----------------|-----------------
Powershell|2.177.0|2.179.1
NodeTool|0.176.0|0.179.1
CmdLine|2.177.3|2.178.0
Npm|1.178.0|1.179.1
Xcode|5.177.0|5.179.0
Bash|3.177.2|3.179.0

Is there a chance one of these changes to the agent could be causing the issue? Also, in order to verify it wasn't our last change that went out we ran the pipeline again today against the last commit that had a successful build for iOS. It is now failing.

**Area for Triage**:  
Apple, Xcode

**Question, Bug, or Feature?**:  
Bug

**Virtual environments affected**  
- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [X] macOS 10.14
- [ ] macOS 10.15
- [ ] macOS 11.0
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Expected behavior**  
The Xcode task would not fail during build executing the "Install Third Party" script.

**Actual behavior**  
The Xcode task is failing while attempting to execute the "Install Third Party" script.

**Comment by maxim-lobanov** [comment]
Hello @woodrufs , could you please also check and share the image version from both builds (first failed build and last successful). Image version can be found in the logs of first step in job

**Comment by woodrufs** [comment]
Hi @maxim-lobanov ,
Thanks for the quick response.
Original Image Version|New Image Version
----------------------|---------------------
20201106.2                  | 20201214.1

**Comment by maxim-lobanov** [comment]
It is the link to diff between 20201106.2 and 20201214.1 images: https://github.com/actions/virtual-environments/pull/2269/files
On the first sight, nothing related to Xcode were changed on new image.

We can't say anything about versions of tasks Powershell, NodeTool, CmdLine, Npm, Bash, Xcode since we don't have much context with it. It is owned by https://github.com/microsoft/azure-pipelines-tasks

Based on my experience, the issue can be caused by updating of some NPM packages. You can try to compare logs of successful and failed builds and compare the versions of packages that "npm install" downloads.

Also suggest to delete "node_modules" folder locally and try to run "npm install" again, probably, you could face with the same issue.

**Comment by ajhall** [comment]
@woodrufs and I got this figured out on our end - it was related to the OpenSSL upgrade on the build agent.

We had a step in our pipeline that explicitly removed OpenSSL 1.0, wiped its config dirs, and installed OpenSSL 1.1 via Homebrew. On the updated Mac image OpenSSL 1.1 is already installed, so our script was just failing the OpenSSL 1.0 uninstall, wiping the OpenSSL config directories, and never replacing them. This was ultimately causing causing curl to fail inside [ios-install-third-party.sh](https://github.com/facebook/react-native/blob/v0.59.10/scripts/ios-install-third-party.sh).

We removed our custom OpenSSL steps and everything seems to be working again.
